### PR TITLE
Bubble sorting algorithm

### PR DESCRIPTION
**Pull request by GuNner-pete**
-This algorithm was designed using Ruby.
-It can sort integer inputs and string ones.

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    while element < (array.length - 1)
+      if array[element] > array[element + 1]
+        array[element], array[element + 1] = array[element + 1], array[element]
+        sorted = false
+      end
+      element += 1
+    end
+  end
+  print array
+end
```
Comment (ebeagusamuel) [inline bubble_sort.rb]: - [x] You should be returning `array` not printing it. Remember to use ruby's implicit return when updating your code.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    while element < (array.length - 1)
+      if (yield array[element], array[element + 1]).to_i >= 0
+        array[element], array[element + 1] = array[element + 1], array[element]
+        sorted = false
+      end
+      element += 1
+    end
+  end
+  print array
+end
```
Comment (ebeagusamuel) [inline bubble_sort.rb]: - [x] same as above

**Review by ebeagusamuel — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
### Hey there :wave: 
#### Nice implementation done on this project, however, it doesn't pass the test cases because of some minor issue which I have explained in the comments below. Please check my comments and make changes accordingly.
- [x] Please update your readme to include how to clone, set up, and run the project locally. 
- [x] Please delete the commented code you have. Do not commit commented code.
### Do request another review once you have made the changes.
slack(@ebeaguusamuel)

**Comment by jubaan** [comment]
# Status: Changes Required :recycle: 

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [ ] These previous observations pending
  - [ ]  Please update your readme to include how to clone, set up, and run the project locally.

## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

**Review by jubaan — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [x] These previous observations pending
  - [x]  Please update your readme to include how to clone, set up, and run the project locally.

## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

**Comment by mimipeshy** [comment]
## STATUS :CHANGES REQUIRED :red_circle: :recycle: 

Hey team @GuNner-pete and @luisvinicius09 :wave: :wave: good job working on this project, it's almost complete.
However, before approving this project, the README file needs a little more work. I have broken down what needs to be done below :arrow_down: 

##### Usage section

- [x] Download/Clone the repository -> This statement is vague for someone who is new to git. To make it more clearer you can give the actual git commands like this `git clone https://github.com/luisvinicius09/bubble_sort.git` 

- [x] You will need to call the function in the end of your local file.
Ex: bubble_sort([3, 5, 7, 19, 293, 0])
or bubble_sort_by(["Your name", "Hey", "hi", "What's up?"]) do |x,y| x.length - y.length end
--> Is this line really necessary?

- [x] You can run the file in the terminal using the command "ruby bubble_sort.rb" -> this line is also vague to someone who is new to git. Does the user need to `cd` into a folder? Add that command. Then give the instruction on how to run your file which is where the above instruction comes into play.
- [x] Or you can copy & paste the code in the terminal and run it. -> This instruction is not necessary 

- [x] In the author's section, there is a typo `Find Me Arround web:` it should be `Find Me Around the web:` you can use [Grammarly ](https://www.grammarly.com/)to check for typos in your README.

That's all :smile: , kindly submit another review when done.

If you have any questions, kindly slack me :smiley: 

Regards
Peris Ndanu

**Review by arslanbisharat — APPROVED** [review]
# Approved :smile: 

Hello Developer :wave:

I just reviewed your project and I am very happy to inform you that your project is approved.
Congratulations on achieving this milestone in your career. :clap:


_I am looking forward to seeing your next project._

Stay safe and give me a smile 😇

### Happy Coding
------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

**Pull request merged**